commonutils/src/main/java/io/intrepid/commonutils/StringUtils.java

```diff
@@ -0,0 +1,504 @@
+ * A utility class bundling a set of handy to use methods that deal with {@link String},
+ * {@link CharSequence}, {@link android.text.Spannable}, etc.
+ * <p/>
+ * Created by Anton Spaans on 4/15/16.
+ */
+@SuppressWarnings("unused")
+public abstract class StringUtils {
+    private static final ThreadLocal<Pattern[]> MATCHING_PATTERNS = new ThreadLocal<Pattern[]>() {
+        @Override
+        protected Pattern[] initialValue() {
+            return new Pattern[] {
+                    Pattern.compile("[\\t\\n\\u0590-\\uFFFF]"), // non-boring characters.
+                    Pattern.compile("[^a-zA-Z0-9]"), // non-alpha-numeric.
+                    Pattern.compile("^([a-z0-9A-Z]+[-_|\\.]?)+[a-z0-9A-Z]@([a-z0-9A-Z]+(-[a-z0-9A-Z]+)?\\.)+[a-zA-Z]{2,}$") // Email-address pattern.
+            };
+        }
+    };
+
+    private static final ThreadLocal<char[]> LOCAL_CHAR_ARRAY_BUFFER = new ThreadLocal<char[]>() {
+        @Override
+        protected char[] initialValue() {
+            return new char[8192];
+        }
+    };
+
+    /**
+     * Removes any trailing and leading whitespaces from the input characters and returns
+     * the resulting characters without trailing and leading whitespaces.
+     *
+     * @param s   Input characters.
+     * @param <T> Type of the input characters.
+     * @return The stripped characters.
+     */
+    public static <T extends CharSequence> T stripSurroundingWhiteSpace(T s) {
+        if (TextUtils.isEmpty(s)) {
+            return s;
+        }
+
+        final int len = s.length();
+
+        int startIdx = -1;
+        for (int idx = 0; idx < len; idx++) {
+            if (!Character.isWhitespace(s.charAt(idx))) {
+                startIdx = idx;
+                break;
+            }
+        }
+
+        if (startIdx < 0) {
+            //noinspection unchecked
+            return (T) ((s instanceof String) ? "" : s.subSequence(0, 0));
+        }
+
+        int endIdx = startIdx - 1;
+        for (int idx = len - 1; idx > endIdx; idx--) {
+            if (!Character.isWhitespace(s.charAt(idx))) {
+                endIdx = idx;
+                break;
+            }
+        }
+
+        //noinspection unchecked
+        return (T) s.subSequence(startIdx, endIdx + 1);
+    }
+
+    /**
+     * Replaces non-boring character with the given replacement string.
+     * Note that the replacement string must contain only boring characters.
+     *
+     * @param string      String whose non-boring character will be replace with the given replacement.
+     * @param replacement The replacement.
+     * @return The string without boring characters.
+     * @see android.text.BoringLayout
+     * <p/>
+     * Boring characters are all character less than \u0590 and not a tab
+     * and not a new line.
+     */
+    public static String replaceNonBoringCharacters(@NonNull CharSequence string, @NonNull String replacement) {
+        return MATCHING_PATTERNS.get()[0].matcher(string).replaceAll(replacement);
```

Comment:
This method would be mainly used to strip out non-boring character (i.e. replacement=""). 
I'll have the caller make sure that the replace is non-boring as well.